packages/emitter-framework/src/python/components/function-declaration/function-declaration.tsx

```diff
@@ -1,37 +1,115 @@
+import { useTsp } from "#core/index.js";
+import { buildParameterDescriptors } from "#python/utils/operation.js";
+import { declarationRefkeys } from "#python/utils/refkey.js";
+import { type Children, List } from "@alloy-js/core";
 import * as py from "@alloy-js/python";
 import type { Model, Operation } from "@typespec/compiler";
+import { TypeExpression } from "../type-expression/type-expression.jsx";
 
 export interface FunctionDeclarationPropsWithType
   extends Omit<py.FunctionDeclarationProps, "name"> {
   type: Operation;
   name?: string;
   parametersMode?: "prepend" | "append" | "replace";
+  parametersModel?: Model;
 }
 
 export type FunctionDeclarationProps =
   | FunctionDeclarationPropsWithType
   | py.FunctionDeclarationProps;
 
 /**
- * A TypeScript function declaration. Pass the `type` prop to create the
+ * Normalize various doc sources into a Python FunctionDoc element.
+ *
+ * - When `source` is a string, it is split into lines and rendered as a multi-line
+ *   docstring (one line per row).
+ * - When `source` is an array of Children, it is rendered as distinct paragraphs.
+ * - When `source` is already a JSX node (e.g., an explicit <py.FunctionDoc />), it is
+ *   returned as-is.
+ */
+function createDocElement($: ReturnType<typeof useTsp>["$"], source: any): Children | undefined {
```

Comment:
Does `source` have to be `any` or can it be `string | Children[] | FunctionDoc` as you describe in the comments above?